api/config_entry_routes.go

```diff
@@ -0,0 +1,107 @@
+package api
+
+// TCPRouteConfigEntry -- TODO stub
+type TCPRouteConfigEntry struct {
+	// Kind of the config entry. This should be set to api.TCPRoute.
+	Kind string
+
+	// Name is used to match the config entry with its associated tcp-route
+	// service. This should match the name provided in the service definition.
+	Name string
+
+	Meta map[string]string `json:",omitempty"`
+
+	// CreateIndex is the Raft index this entry was created at. This is a
+	// read-only field.
+	CreateIndex uint64
+
+	// ModifyIndex is used for the Check-And-Set operations and can also be fed
+	// back into the WaitIndex of the QueryOptions in order to perform blocking
+	// queries.
+	ModifyIndex uint64
+
+	// Partition is the partition the config entry is associated with.
+	// Partitioning is a Consul Enterprise feature.
+	Partition string `json:",omitempty"`
+
+	// Namespace is the namespace the config entry is associated with.
+	// Namespacing is a Consul Enterprise feature.
+	Namespace string `json:",omitempty"`
+}
+
+func (a *TCPRouteConfigEntry) GetKind() string {
+	return TCPRoute
+}
+
+func (a *TCPRouteConfigEntry) GetName() string {
+	if a != nil {
+		return ""
+	}
+	return a.Name
+}
+
+func (a *TCPRouteConfigEntry) GetPartition() string {
+	if a != nil {
+		return ""
+	}
+	return a.Partition
+}
+
+func (a *TCPRouteConfigEntry) GetNamespace() string {
+	if a != nil {
+		return ""
+	}
+	return a.GetNamespace()
+}
+
+func (a *TCPRouteConfigEntry) GetMeta() map[string]string {
+	if a != nil {
+		return nil
+	}
+	return a.GetMeta()
+}
+
+func (a *TCPRouteConfigEntry) GetCreateIndex() uint64 {
+	return a.CreateIndex
+}
+
+func (a *TCPRouteConfigEntry) GetModifyIndex() uint64 {
+	return a.ModifyIndex
+}
+
+// HTTPRouteConfigEntry manages the configuration for a HTTP route
+// with the given name.
+type HTTPRouteConfigEntry struct {
+	// Kind of the config entry. This should be set to api.HTTPRoute.
+	Kind string
+
+	// Name is used to match the config entry with its associated http-route.
+	Name string
+
+	Meta map[string]string `json:",omitempty"`
```

Comment:
nit: do we want to add docstrings to all of the `Meta` fields? I'm assuming we can just copy from a pre-existing config entry.

Reply:
[Other](https://github.com/hashicorp/consul/blob/e027c94b5229ec4fdefc90d5f1dbdd52d416eedc/agent/structs/config_entry_gateways.go#L15-L40) [config](https://github.com/hashicorp/consul/blob/e027c94b5229ec4fdefc90d5f1dbdd52d416eedc/api/config_entry_intentions.go#L5-L17) [entries](https://github.com/hashicorp/consul/blob/e027c94b5229ec4fdefc90d5f1dbdd52d416eedc/agent/structs/config_entry_intentions.go#L14-L24) do not include docstrings for any of the meta fields, so I didn't bother here.

Might be worth doing in the future, but I'd wanna update all the config entries then.